### PR TITLE
engine/utils: refactor the function compare_string_view()

### PR DESCRIPTION
**Pull request by jinhao2**
### What problem does this PR solve?
Problem Summary:
compare_string_view() function works inefficiently by comparing characters one by one .

### What is changed and how it works?
When the Key is more than 8B,  compare two uint64_t instead of  8 bytes.

What's Changed:
static inline int compare_string_view(const StringView& src, const StringView& target) {
  auto size = std::min(src.size(), target.size());
  auto batchNum = size & ~(sizeof(uint64_t)-1);     
  uint32_t i = 0;
  for( i =0; i< batchNum ;  ){      // if size is more than 8, compared by type of  uint64_t.
      auto nEq = *(uint64_t*)(&src[i]) - *(uint64_t*)(&target[i]);
      if (  nEq  )
		return nEq;
    i += sizeof(uint64_t);
  }
  for (; i < size; i++) {
    if (src[i] != target[i]) {
      return src[i] - target[i];    
	}
  }
  return src.size() - target.size();
}

Tests 
- No code

Side effects
none

### CODE REVIEW

**Review by JiayuZzz — APPROVED** [review]
Thank, LGTM

**Comment by iyupeng** [comment]
C++ code style can be fixed by:
```
clang-format --version
# clang-format version 9.0.0
clang-format --style=file -i engine/utils/utils.hpp
```

**Comment by iyupeng** [comment]
Hi @jinhao2 comparing string with `uint64_t` may be problematic since the Little Endian memory layout on x86 platforms. We can use `uint64_t` to check equality but not to compare strings.

By the way, `compare_string_view` is supposed to produce results based on lexicographical order. So bytes should be interpreted as `unsigned char`. And we fixed this by: https://github.com/pmem/kvdk/pull/340

In some implementations, `memcmp` or `strcmp` is applied to compare strings. But we choose to keep the current `for-loop` for now based on benchmark results.

File: engine/utils/utils.hpp [hunk 1]
```diff
@@ -153,7 +153,15 @@ inline std::string string_view_2_string(const StringView& src) {
 inline int compare_string_view(const StringView& src,
                                const StringView& target) {
   auto size = std::min(src.size(), target.size());
-  for (uint32_t i = 0; i < size; i++) {
+  auto batchNum = size & ~(sizeof(uint64_t)-1);
+  uint32_t i = 0;
+  for( i =0; i< batchNum ; ){
+    auto nEq = *(uint64_t*)(&src[i]) - *(uint64_t*)(&target[i]);
+    if (nEq)
```
Comment (ZiyanShi) [inline engine/utils/utils.hpp:160]: The algorithm is incorrect.
x86 systems adopts little endianness. In other words, you have to reverse 8 bytes first, then treat them as a uint64_t for comparison.
Also, when uint64_t converts to int, overflow may occur and the result will be wrong. For example, `"Hello, World!"` and `"Hello, world!"` will be equal under this algorithm.

Comment (jinhao2) [reply engine/utils/utils.hpp:160]: OK. I thought it was only used for equivalence compare.
On little endian host the return value is the opposite.

**Pull request closed without merging**